### PR TITLE
Use `FlaggedStorage` for `Transform` component.

### PR DESCRIPTION
**Pull request by Rhuagh**
`TransformSystem`will clear the flags. If `TransformSystem` is not used, the user is in charge of clearing the flags.

### CODE REVIEW

File: src/ecs/transform/systems.rs [hunk 1]
```diff
@@ -76,10 +80,13 @@ impl<'a> System<'a> for TransformSystem {
 
         {
             // Compute transforms without parents.
-            for (ent, local, global, _) in (&*entities, &locals, &mut globals, !&children).join() {
+            for (ent, local, _, _) in (&*entities, &locals, &globals.check(), !&children).join() {
```
Comment (Aceeri) [inline src/ecs/transform/systems.rs]: Probably use `RestrictedStorage` here to avoid checks.

File: amethyst_assets/Cargo.toml [hunk 2]
```diff
@@ -25,5 +25,5 @@ fnv = "1"
 futures = "0.1.14"
 log = "0.3.8"
 parking_lot = "0.4.4"
-rayon = "0.7"
-specs = { version = "0.9.5", features = ["common"] }
+rayon = "0.8"
+specs = { version = "0.10", features = ["common"] }
```
Comment (torkleyy) [inline amethyst_assets/Cargo.toml]: We should also use the `serde` feature, which is what @omni-viral implemented

Comment (torkleyy) [reply amethyst_assets/Cargo.toml]: No reason to stop bors ^^

**Pull request merged**